KUBERNETES.md

```diff
@@ -0,0 +1,180 @@
+  * `DeleteInBackground` — will delete the referenced object immediately. All its descendants will be removed by Kubernetes'
+    garbage collector.
+  * `DeleteNonCascading` — will delete the referenced object immediately, and orphan all its descendants.
+  apiVersion from `discovery` client.
+* `apiVersion` — optional field that specifies object's apiVersion. If not present, we'll use preferred apiVersion
+  for the given kind.
+* `kind` — object's Kind.
+* `namespace` — object's namespace. If empty, implies operation on a cluster-level resource.
+* `name` — object's name.
+* `subresource` — e.g., `status`.
+
+#### Example
+
+```json
+{
+  "operation": "Delete",
+  "kind": "Pod",
+  "namespace": "default",
+  "name": "nginx"
+}
+```
+
+### Patch
+
+Use `JQPatch` for almost everything. Consider using `MergePatch` or `JSONPatch` if you are attempting to modify 
+rapidly changing object, for example `status` field with many concurrent changes (and incrementing `resourceVersion`).
+
+#### JQPatch
+
+* `operation` — specifies an operation's type.
+* `apiVersion` — optional field that specifies object's apiVersion. If not present, we'll use preferred apiVersion
+  for the given kind.
+* `kind` — object's Kind.
+* `namespace` — object's Namespace. If empty, implies operation on a Cluster-level resource.
+* `name` — object's name.
+* `jqExpression` — describes transformations to perform on an object.
+* `subresource` — e.g., `status`.
+
+##### Example
+
+```json
+{
+  "operation": "JQPatch",
+  "kind": "Deployment",
+  "namespace": "default",
+  "name": "nginx",
+  "jqExpression": ".spec.replicas = 1"
+}
+```
+
+#### MergePatch
+
+* `operation` — specifies an operation's type.
+* `apiVersion` — optional field that specifies object's apiVersion. If not present, we'll use preferred apiVersion
+  for the given kind.
+* `kind` — object's Kind.
+* `namespace` — object's Namespace. If empty, implies operation on a Cluster-level resource.
+* `name` — object's name.
+* `mergePatch` — describes transformations to perform on an object.
+* `subresource` — e.g., `status`.
+
+##### Example
+
+```json
+{
+  "operationType": "MergePatch",
+  "kind": "Deployment",
+  "namespace": "default",
+  "name": "nginx",
+  "mergePatch": "{\"spec\":{\"replicas\":1}}"
+}
+```
+
+#### JSONPatch
+
+* `operation` — specifies an operation's type.
+* `apiVersion` — optional field that specifies object's apiVersion. If not present, we'll use preferred apiVersion
+  for the given kind.
+* `kind` — object's Kind.
+* `namespace` — object's Namespace. If empty, implies operation on a Cluster-level resource.
+* `name` — object's name.
+* `jsonPatch` — describes transformations to perform on an object.
+* `subresource` — e.g., `status`.
+
+##### Example
+
+```json
+{
+  "operationType": "JSONPatch",
+  "kind": "Deployment",
+  "namespace": "default",
+  "name": "nginx",
+  "jsonPatch": "[{\"op\":\"replace\",\"path\":\"/spec/replicas\",\"value\":1}]"
```

Comment:
Same thought about this field as in https://github.com/flant/shell-operator/pull/239#discussion_r548863035
Maybe we can use an object as the value of `jsonPatch` and `mergePatch` fields.

Reply:
Done.